src/charts/time_series_labeler.coffee

```diff
@@ -56,112 +85,156 @@ Ember.Charts.TimeSeriesLabeler = Ember.Mixin.create
       ticks
     else
       count = 1
-      interval = switch @get('selectedInterval')
-        when 'years', 'Y' then 'year'
-        when 'quarters', 'Q' then 'quarter'
-        when 'months', 'M' then 'month'
-        when 'weeks' , 'W'  then 'week'
-        when 'hours' , 'H' then 'hour'
-        when 'seconds', 'S' then 'second'
+      interval = switch @get('xAxisDomainType')
```

Comment:
why not use `domainTypeToLabellerType`? Oh I suppose you want singular :/ maybe make another one? I know it seems a little ridiculous but it would make the actual logic really concise :+1: